source/storing-data/archive-versions.rst

```diff
@@ -0,0 +1,226 @@
+Archive Versions
+================
+.. contents::
+   :local:
+
+As QIIME 2 has developed, the structure of QIIME 2 :term:`archives<Archive>` has evolved.
+This page describes each historical version of the QIIME 2 Archive format,
+and may be useful to developers whose code depends on guarantees made by that format
+(`source code <https://github.com/qiime2/qiime2/blob/master/qiime2/core/archive/>`_).
+
+For general information about the structure of current QIIME 2 Archives, see :doc:`archive`.
+For a detailed description of the part of an Archive which holds Provenance data, see :doc:`provenance`.
+
+Version-agnostic format "guarantees"
+------------------------------------
+
+Though there is significant variability in the format of QIIME 2 Archives across versions,
+all versions share some common traits.
+
+These shared characteristics, defined in the ``_Archive`` class
+in `qiime2/core/archive/archiver.py <https://github.com/qiime2/qiime2/blob/master/qiime2/core/archive/archiver.py>`_,
+must be consistent across all formats over time, 
+as they allow archive versions to be checked,
+and archives with different formats to be dispatched to the appropriate version-specific tools.
+
+All QIIME 2 Archives have:
+
+- a directory named with the Archive UUID, directly under the archive root.
+- a file named "VERSION" within that directory, formatted as shown below
+
+The Archive file system must look like this:
+
+.. figure:: ../img/archive_versions/format_agnostic_archive_structure.svg
+   :alt: Box and Arrow diagram of the guaranteed components of an archive, as described above.
+
+The VERSION file must look like this::
+
+    QIIME 2
+    archive: <archive version>
+    framework: <framework version>
+
+.. note::
+   This file is itentionally not YAML/INI/An actual format. This is to
+   discourage the situation where the format changes from something like YAML to
+   another format and VERSION is updated with it "for consistency".
+
+Version 0
+---------
+
+The original QIIME 2 Archive format, there aren't many V0 Archives "in the wild".
+V0 Archives, in place through 10/24/16, were produced only during QIIME 2's Alpha release,
+prior to version 2.0.6.
+
+- Like all QIIME 2 Archives, there is a directory named with the Archive UUID under the archive root, containing a VERSION file. (This is *not* format-defined. See above.)
+- :term:`Result` data is written to a ``data`` directory, located within this UUID directory.
+- Result UUID, Semantic Type, and Format go in a ``metadata.yaml`` in the same place.
+- The ArchiveFormat class in `v0.py <https://github.com/qiime2/qiime2/blob/master/qiime2/core/archive/format/v0.py>`_ offers convenience methods for loading and parsing ``metadata.yaml`` files.
+
+V0 Archives look like this:
+
+.. figure:: ../img/archive_versions/v0_archive_fmt.svg
+   :alt: Box and Arrow diagram of a v0 archive, as described above.
+
+Version 1
+---------
+
+Created in `PR #171 <https://github.com/qiime2/qiime2/pull/171>`_,
```

Comment:
Maybe we should link to the commit URL on `master`, instead of the PR URL - that commit is technically when it was added into QIIME 2 - PRs are just a tool of GitHub, but the commit seems like the canonical representation of the change. This is me just thinking out loud though, I don't really care either way.

Reply:
I'm super torn on this one! :laughing: I think you're absolutely right, that the commit is "real", while the PR is interface-dependent. I chose PRs because that they often offer extra context - the discussion and closed issues like we have [here](https://github.com/qiime2/qiime2/pull/387) were helpful for me in writing these up. 

What do you think the odds are we'd move away from GH at some point? If  that seemed likely, I'd be very inclined to go with commits, just to keep things consistent for future readers. You can get to all the nice-to-haves above from the commit, it's just one click farther.

Reply:
> What do you think the odds are we'd move away from GH at some point?

Very low.

> just to keep things consistent for future readers

One note, while not common at all, some features might not be expressed by PRs at all, we have been known to merge-to-master on occassion.

Reply:
Fair point. Sold!